SmallBetterLogger/SmallBetterLogger.hpp

```diff
@@ -0,0 +1,179 @@
+		}
+
+		// Copy constructor
+		Logger(const Logger& other)
+			: m_StreamType(other.m_StreamType), m_AutoFlush(other.m_AutoFlush)
+		{ 
+			if (other.m_StreamType == STREAM_TYPE::FILE) throw std::exception("Cannot make a copy of a logger which outputs to a file stream.");
+		}
+
+		// Flush and close file stream if open
+		~Logger()
+		{
+			if (m_FileStream.is_open())
+			{
+				m_FileStream.flush();
+				m_FileStream.close();
+			}
+		}
+
+		// Public Methods
+
+		// Writes to the stream a message and inserts values into placeholders (should they exist)
+		template<typename ...T>
+		inline void Write(const std::string& message, const T& ...t);
+
+		// Writes to the stream a message and inserts values into placeholders (should they exist)
+		template<typename ...T>
+		inline void Write(const std::string&& message, const T&& ...t);
+
+		// Writes to the stream a message and inserts values into placeholders (should they exist) and finishes with the newline character
+		template<typename ...T>
+		inline void WriteLine(const std::string& message, const T& ...t);
+
+		// Writes to the stream a message and inserts values into placeholders (should they exist) and finishes with the newline character
+		template<typename ...T>
+		inline void WriteLine(const std::string&& message, const T&& ...t);
+
+	private:
+		// Private members
+		STREAM_TYPE m_StreamType;
+		std::fstream m_FileStream;
+		bool m_AutoFlush;
+
+		// Private methods
+
+		// Converts a T value to a string to be used in writing a log
+		template<typename T>
+		inline std::string stringConvert(const T& t);
+
+		// Replace all "{n}" placeholders with their respective values (n=0,...)
+		inline std::string replacePlaceholders(std::string message, std::vector<std::string>& items);
+
+		// Writes string to appropriate stream based on instance STREAM_TYPE (m_StreamType)
+		inline void writeToStream(const std::string&& str);
+	};
+
+	template<typename ...T>
+	inline void Logger::Write(const std::string& message, const T& ...t)
+	{
+		std::vector<std::string> printValues { stringConvert(t)... };
+		writeToStream(replacePlaceholders(message, printValues));
+	}
+
+	template<typename ...T>
+	inline void Logger::Write(const std::string&& message, const T&& ...t)
+	{
+		std::vector<std::string> printValues{ stringConvert(t)... };
+		writeToStream(replacePlaceholders(message, printValues));
+	}
+
+	template<typename ...T>
+	inline void Logger::WriteLine(const std::string& message, const T& ...t)
+	{
+		std::vector<std::string> printValues{ stringConvert(t)... };
+		writeToStream(replacePlaceholders(message, printValues) + "\n");
+	}
+
+	template<typename ...T>
+	inline void Logger::WriteLine(const std::string&& message, const T&& ...t)
+	{
+		std::vector<std::string> printValues{ stringConvert(t)... };
+		writeToStream(replacePlaceholders(message, printValues) + "\n");
+	}
+
+	template<typename T>
+	inline std::string Logger::stringConvert(const T& t)
```

Comment:
Stick to a naming convention. All methods above follow PascalCase, all methods below folow CamelCase. Make up your mind.